src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcEndpointDataSource.cs

```diff
@@ -101,7 +109,48 @@ private void InitializeEndpoints()
                     action.RouteValues,
                     action.AttributeRouteInfo.Order,
                     metadataCollection,
-                    action.DisplayName));
+                    action.DisplayName,
+                    new Address(action.AttributeRouteInfo.Name)));
+            }
+
+            foreach (var info in _options.Value.Endpoints)
```

Comment:
nit: info -> endpointInfo

Reply:
As discussed offline, I will not check this in. It was to experiment the integration of our current functional tests over dispatcher.